swarm.go

```diff
@@ -111,8 +119,18 @@ func NewSwarm(ctx context.Context, local peer.ID, peers peerstore.Peerstore, bwc
 	s.limiter = newDialLimiter(s.dialAddr)
 	s.proc = goprocessctx.WithContextAndTeardown(ctx, s.teardown)
 	s.ctx = goprocessctx.OnClosingContext(s.proc)
+	s.eventBus = eventbus.NewBus()
```

Comment:
Ideally, libp2p would pass this in on construction.

Well, ideally we'd merge the swarm/host but that's a larger refactor.